ui/segmented-bar.reel/segment.reel/segment.js

```diff
@@ -0,0 +1,93 @@
+/**
+ *@module ui/segmented-bar.reel/segment.reel
+ *@requires montage/ui/component
+ */
+var Component = require("montage/ui/component").Component;
+
+/**
+ * @class Segment
+ * @extends Component
+ * A Segment composes of data that forms a SegmentedBar. A Segment
+ * is bound to an object from SegmentedBar's content array. An array will be projected
+ * as a segment. The 'contentController' (rangeController/treeController)
+ * will use the proper logic and project the array of the rangeController or
+ * flatten the treeController data. The data projected will draw itself
+ * based on a percentage of the bar's total length. The remaining space
+ * will be its own Segment in the SegmentedBar.
+ */
+exports.Segment = Component.specialize({
+    label: {value: null},
+    _value: {value: null},
+    value: {
+        get: function () {return this._value},
+        set: function (value) {
+            this._value = value;
```

Comment:
You can maybe check if the new value is different from the old one in order to avoid some process.